### ISSUE TITLE
Device type "ct" (Farbtemperatur) - multiple problems with state "ON"

### ISSUE THREAD

**Issue by klein0r** [ioBroker/ioBroker.devices#159]
**Describe the bug**  

- Create a new device of type ``ct``
- Fill temperature, dimmer and on
- Save the device

**Issues**

1. Object ``ON`` has no role (should be ``switch.light``)
2. Reopen the device in devices tab: input field for ``ON`` is empty (but reference exists)
3. Delete the device in devices tab: ``ON`` isn't deleted

**Versions:**  
 - Adapter version: 1.0.10

**Comment by Garfonso** [comment]
I stumbled upon the same issue. The problem is, that the State-Object ON is created without a role. 

This probably has to do with two possible `ON` members in the type-detector pattern:
```
                {role: /^switch\.light$/,                                 indicator: false, type: 'boolean', write: true,           name: 'ON',            required: false, defaultRole: 'switch.light'},
                {role: /^switch$/,                                        indicator: false, type: 'boolean', write: true,           name: 'ON',            required: false},
```

The second one has no `defaultRole` defined. But this seems necessary, because items without a `defaultRole` get filtered out. I.e. if a `defaultRole` was set on both patterns, `ON` would show up two times in the states list (which happens on some devices, RGB-single for example). 

I did not yet understand how the devices-adapter decides which role to use on object creation and how that works if there are two patterns with the same name. Furthermore, I tested a few other devices (airCondition) and it seems to happen on all of them. 

For ct a fix in type-detector is easy (just make the regex more complex) but for other devices two patterns with the same name can be necessary (for example for accepting different types).

**Comment by klein0r** [comment]
No!! Still an issue 😢

**Comment by klein0r** [comment]
⬆️

**Comment by Garfonso** [comment]
are you sure that this is still an issue? Can you test with version 1.1.2, please? I can currently not reproduce this issue here.

**Comment by Garfonso** [comment]
It should already be fixed in 1.1.2. 

With 1.1.3 I also have the blank page issue. I fixed that in my fork (and my PR is still open). You can try to install from https://github.com/Garfonso/ioBroker.devices

**Comment by klein0r** [comment]
✅  (Tested with 1.1.5)
